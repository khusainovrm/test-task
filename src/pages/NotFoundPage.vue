<template>
  <div class="h-dvh grid place-items-center">
    <div class="flex flex-col gap-4 items-center">
      <span class="text-4xl">Not found</span>
      <button @click="toMain">To main page</button>
      <span class="text-9xl">ğŸ‘¹</span>
    </div>
  </div>
</template>

<script setup lang="ts">
import { useRouter } from 'vue-router';

const router = useRouter();

const toMain = () => {
  router.push('/');
};
</script>

<style scoped></style>
