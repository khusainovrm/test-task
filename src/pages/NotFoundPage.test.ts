import { describe, expect, test, vi } from 'vitest';
import { useRouter } from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import NotFoundPage from './NotFoundPage.vue';

vi.mock('vue-router');

describe('NotFounePage', () => {
  (useRouter as any).mockReturnValue({
    push: vi.fn(),
  });

  test('mount component', async () => {
    expect(NotFoundPage).toBeTruthy();

    const wrapper = shallowMount(NotFoundPage);

    expect(wrapper.text()).toContain('Not found');
    expect(wrapper.html()).toMatchSnapshot();

    await wrapper.get('button').trigger('click');

    expect(useRouter().push).toHaveBeenCalledWith('/');
  });
});
