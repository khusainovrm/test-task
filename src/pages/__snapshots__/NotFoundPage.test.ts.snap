// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFounePage > mount component 1`] = `
"<div class="h-dvh grid place-items-center">
  <div class="flex flex-col gap-4 items-center"><span class="text-4xl">Not found</span><button>To main page</button><span class="text-9xl">ğŸ‘¹</span></div>
</div>"
`;
